etl/src/main/java/org/pih/hivmigration/etl/sql/ExamLabResultsMigrator.groovy

```diff
@@ -243,6 +240,61 @@ class ExamLabResultsMigrator extends ObsMigrator {
         ''')
 
         migrate_tmp_obs()
+
+        //
+        // -------- Hemoglobin (and some hematocrit)
+        //
+
+        create_tmp_obs_table()
+
+        executeMysql("Migrate hemoglobin values", '''
+            INSERT INTO tmp_obs (source_encounter_id, concept_uuid, value_numeric)
+            SELECT
+                he.source_encounter_id,
+                concept_uuid_from_mapping('PIH', 'HEMOGLOBIN'),
+                extract_number(result)
+            FROM hivmigration_exam_lab_results helr
+            JOIN hivmigration_encounters he on helr.source_encounter_id = he.source_encounter_id
+            JOIN encounter e on he.encounter_id = e.encounter_id
+            WHERE lab_test = 'hemoglobin'
+                AND (result REGEXP '^[[:blank:]]*[0-9\\.]+[[:blank:]]*gr?.dll?$'  -- e.g. '9.8 g/dl' or '9.8gr/dl'
+                     OR is_number(result))  -- e.g. 9.8
+            ;
+        ''')
+
+        executeMysql("Migrate hematocrit values incorrectly marked as hemoglobin", '''
+            INSERT INTO tmp_obs (source_encounter_id, concept_uuid, value_numeric)
+            SELECT
+                he.source_encounter_id,
+                concept_uuid_from_mapping('PIH', 'HEMATOCRIT'),
+                extract_number(result)
+            FROM hivmigration_exam_lab_results helr
+            JOIN hivmigration_encounters he on helr.source_encounter_id = he.source_encounter_id
+            JOIN encounter e on he.encounter_id = e.encounter_id
+            WHERE lab_test = 'hemoglobin'
+                AND result REGEXP '^(Hte?|HTE?)?[[:blank:]]?[0-9\\.]+%$'  -- e.g. '33%' or 'Ht 33.0%'
```

Comment:
Seems a reasonable assumption to me, but don't know if we need any outside validation?

Reply:
Probably would be good— @dmdesimone or @lnball ?

Reply:
The question is— given a bunch of values mostly in the range 25-40, which are given as percentages, even though they were put into the "hemoglobin" box can we assume that they're actually hematocrit?

Reply:
Resolved in a chat in Teams—we're just going to drop those values (they'll get picked up by the warnings table).